lib/common/json-schema-validator.js

```diff
@@ -90,25 +99,116 @@ function jsonSchemaValidatorFactory(
+                '`name` should not be empty, or `schema` should has an `id` property');
+
+        schema.id = url.resolve(this.nameSpace, name || schema.id);
+        return this._ajv.addSchema(schema);
+    };
+
+    /**
+     * Add array of schemas to the instance.
+     * @param {Array<Object>} schemas - array of schema
+     */
+    JsonSchemaValidator.prototype.addSchemas = function (schemas) {
+        assert.arrayOfObject(schemas,
+                '`schemas` should be an array of JSON schema');
+        var self = this;
+        _.forEach(schemas, function (schema) {
+            self.addSchema(schema);
+        });
+    };
+
+    /**
+     * Add all schema in the folder to the instance.
+     * @param {String} schemaDir - the directory where schema files in
+     * @param {String} [metaSchemaName] - Optional the meta schema name
+     * @return {Promise}
+     */
+    JsonSchemaValidator.prototype.addSchemasByDir = function (schemaDir, metaSchemaName) {
+        var self = this;
+        return Promise.try(function () {
+            assert.string(schemaDir, 'schemaDir');
+            if (metaSchemaName) {
+                assert.string(metaSchemaName, 'metaSchemaName');
+            }
+        })
+        .then(function () {
+            return fs.readdirAsync(schemaDir);
+        })
+        .filter(function (fileName) {
+            return /\.json$/i.test(fileName);
+        })
+        .reduce(function (entries, fileName) {
+            return fs.readFileAsync(path.resolve(schemaDir, fileName))
+            .then(function (fileData) {
+                var content = JSON.parse(fileData);
+                content.id = fileName;
+                entries[fileName] = content;
+                return entries;
+            });
+        }, {})
+        .then(function (entries) {
+            //meta schema should be added first, so subsequent schemas can be validated againist it
+            if (metaSchemaName) {
+                var metaSchema = entries[metaSchemaName];
+                if (!metaSchema) {
+                    throw new Error('Cannot find the meta schema "' + metaSchemaName +
+                        '" in directory "' + schemaDir + '"');
+                }
+                self.addMetaSchema(metaSchema);
+                delete entries[metaSchemaName];
+            }
+            return entries;
+        })
+        .then(function (entries) {
+            self.addSchemas(_.values(entries));
+        });
+    };
+
+    /**
+     * Get all schema names from the instance.
+     * @param {Object} [option] - namespace will not be included by default,
+     *      set { includeNameSpace: true } when need to include namespace.
+     * @return {Array<String>} schemaNames
+     */
+    JsonSchemaValidator.prototype.getAllSchemaNames = function (option) {
+        var self = this;
+        return _(self._ajv._schemas).keys()
+        .filter(function (name) {
+            return name.startsWith(self.nameSpace);
```

Comment:
If `nameSpace` is empty, then the `startsWith` will always true, but this is not expected:
```
> "abc".startsWith('')
true
```